tests/test_expressions.py

```diff
@@ -247,6 +247,16 @@ def fun(node):
 
         self.assertEqual(expression.transform(fun).sql(), "SELECT a, b FROM x")
 
+    def test_transform_node_removal(self):
```

Comment:
Other tests to consider adding:
1. Remove a node that isn't a list arg
2. Remove all nodes from a list arg
3. Remove the root node